### PR TITLE
Making sure that wrong command line options cause non-zero exit code

### LINKED ISSUE #1507: Unrecognized Command Line Option Error causing exit status 0
Minimal Test Code:

```C++
#include <hpx/hpx_init.hpp>

using boost::program_options::variables_map;
using boost::program_options::options_description;

int hpx_main(variables_map & vm)
{
    return hpx::finalize();
}

int main(int argc, char* argv[])
{
    options_description cmdline("Usage: " HPX_APPLICATION_STRING " [options]");
    return hpx::init(cmdline, argc, argv);
}
```

Called on commandline:
```
> ./test -nonsense
runtime_support::load_components: command line processing: unrecognised option '-nonsense'
> echo "$?"
0
```

In my opinion, the program should not exit with 0 in this case.

### PR DESCRIPTION
**Pull request by hkaiser**
This fixes #1507

### CODE REVIEW

**Comment by hkaiser** [comment]
@Finomnis Could you please verify whether this patch solves your issue?

File: src/runtime/components/server/runtime_support_server.cpp [hunk 1]
```diff
@@ -1569,7 +1571,7 @@ namespace hpx { namespace components { namespace server
                     throw hpx::detail::command_line_error(
                         "unknown help option: " + help_option);
                 }
-                return false;
+                return 0;
```
Comment (sithhell) [inline src/runtime/components/server/runtime_support_server.cpp]: This should return -1 or something. It was marked as "failing" before.

Comment (hkaiser) [inline src/runtime/components/server/runtime_support_server.cpp]: Thanks. That's fixed now.

**Comment by Finomnis** [comment]
Fixed the problem I had.

**Pull request merged**